src/webshare.js

```diff
@@ -289,14 +388,131 @@ const webshare = {
                 item.parsedTitle.resolution +
                 "|" +
                 item.parsedTitle.source, //secures quite reliable auto play next episode
+                                      //
               videoSize: item.size, //for subtitle addons
               filename: item.name, //for subtitle addons
             },
 
             queries: [queryTitle, queryTitleOriginal, queryTitleSk],
             parsedTitle: cleanedTitle,
+            // Initialize resolution using 3-level fallback: filename -> API -> file size
+            resolution: determineResolutionFallback(item.name, item.size, null, null),
           };
-        })
+        });
+
+    // Enrich ALL relevant items (strongMatch + weakMatch) with detailed info (getById) to fetch bitrate/width/height
+    try {
+      const relevantIndices = mapped
+        .map((m, idx) => ({ idx, ident: m.ident, relevant: m.strongMatch || m.weakMatch, strong: m.strongMatch, weak: m.weakMatch }))
+        .filter((x) => x.relevant)
+        .map((x) => ({ idx: x.idx, ident: x.ident }));
+
+      console.log(`Enriching ${relevantIndices.length} relevant items out of ${mapped.length} total mapped items`);
+
+      if (relevantIndices.length > 0) {
+        // Limit concurrent requests to avoid overwhelming the API
+        const chunkSize = 20;
+        const chunks = [];
+        for (let i = 0; i < relevantIndices.length; i += chunkSize) {
+          chunks.push(relevantIndices.slice(i, i + chunkSize));
+        }
+
+        let allDetails = [];
+        for (const chunk of chunks) {
+          const chunkDetails = await Promise.all(
+            chunk.map(async (c) => {
+              try {
+                return module.exports.getById ? await module.exports.getById(c.ident, token) : null;
+              } catch (err) {
+                console.log(`getById failed for ${c.ident}:`, err.message);
+                return null;
+              }
+            }),
+          );
+          allDetails = allDetails.concat(chunkDetails);
+        }
+
+        for (let i = 0; i < allDetails.length; i++) {
+          const d = allDetails[i];
+          const targetIdx = relevantIndices[i].idx;
+          if (d && mapped[targetIdx]) {
+            const target = mapped[targetIdx];
+            
+            // attach bitrate and dimensions for display/sorting
+            target.bitrate = d.bitrate || null;
+            target.width = d.width || null;
+            target.height = d.height || null;
+            
+            // Recalculate resolution using three-level fallback with API data
+            target.resolution = determineResolutionFallback(d.filename || target.description, target.fileSize, d.width, d.height);
+            
+            // Update name field to show correct resolution after enrichment
+            target.name = `Webshare${target.strongMatch ? " ✅" : ""} ${target.resolution || ""}`;
```

Comment:
I'd rather do something like 
```
target.name = `${target.name} ${target.resolution || ""}`
```

instead of constructing the name here again. This way you duplicate the name construction to two places.